aria-practices.html

```diff
@@ -3697,6 +3697,1247 @@ <h5>Examples</h5>
+            <dl class="switch">
+              <dt><code>input</code> whose <code>type</code> attribute is in the Button, Submit Button, or Reset Button state</dt>
+              <dd>The <code>value</code> attribute.</dd>
+
+              <dt><code>input</code> whose <code>type</code> attribute is in the Image Button state</dt>
+              <dt><code>img</code></dt>
+              <dt><code>area</code></dt>
+              <dd>The <code>alt</code> attribute. </dd>
+
+              <dt><code>fieldset</code></dt>
+              <dd>The first child <code>legend</code> element.</dd>
+
+              <dt>Other form elements</dt>
+              <dd>The associated <code>label</code> element(s).</dd>
+
+              <dt><code>figure</code></dt>
+              <dd>The first child <code>figcaption</code> element.</dd>
+
+              <dt><code>table</code></dt>
+              <dd>The first child <code>caption</code> element.</dd>
+
+              <dt><code>area</code></dt>
+              <dd>The <code>alt</code> attribute.</dd>
+            </dl>
+          </li>
+          <li><p>If the name is still empty, then for elements with a role that supports naming from child content, the content of the element is used.</p></li>
+          <li>
+            <p>Finally, if the name is still empty, then other fallback host-language-specific attributes or elements are used if present. For HTML, these are, depending on the element:</p>
+            <dl class="switch">
+              <dt><code>input</code> whose <code>type</code> attribute is in the Text, Password, Search, Telephone, or URL states</dt>
+              <dt><code>textarea</code></dt>
+              <dd>The <code>title</code> attribute. Otherwise, the <code>placeholder</code> attribute.</dd>
+
+              <dt><code>input</code> whose <code>type</code> attribute is in the Submit Button state</dt>
+              <dd>A localized string of the word "submit".</dd>
+
+              <dt><code>input</code> whose <code>type</code> attribute is in the Reset Button state</dt>
+              <dd>A localized string of the word "reset".</dd>
+
+              <dt><code>input</code> whose <code>type</code> attribute is in the Image Button state</dt>
+              <dd>The <code>title</code> attribute. Otherwise, a localized string of the phrase "Submit Query".</dd>
+
+              <dt><code>summary</code></dt>
+              <dd>The word "Details".</dd>
+
+              <dt>Other elements</dt>
+              <dd>The <code>title</code> attribute.</dd>
+            </dl>
+          </li>
+        </ol>
+        <p>
+          The final step is a fallback mechanism. Generally when labelling an element, use one of the non-fallback mechanisms.
+        </p>
+        <p>
+          When calculating a name from content, the user agent walks through all descendant nodes.
+          And, when following <code>aria-labelledby</code> references, it similarly walks the tree of each referenced element.
```

Comment:
How can we word this so that it doesn't sound like it may be talking about "aria-labelledby chaining"? I was thinking maybe:
```
And, when following references in an <code>aria-labelledby</code> attribute, it similarly walks the tree of each referenced element.
```